### PR TITLE
strip the uid that crio is injecting

### PR DESCRIPTION
**Pull request by bparees**

### CODE REVIEW

**Comment by bparees** [comment]
/assign @gabemontero @adambkaplan 

let's see if this gets us green......  this will run the test that's failing, right?

**Comment by gabemontero** [comment]
yep it runs the problematic slave pod tests @bparees

**Comment by gabemontero** [comment]
and we have https://github.com/openshift/jenkins/pull/845 that is now seeing this too @bparees @adambkaplan to compare with

**Comment by adambkaplan** [comment]
/hold

If this is required by a recent cri-o update, we may not need this if the update is reverted.

**Comment by gabemontero** [comment]
same error @bparees 

```
WARNING: No Working Directory. Using the legacy JAR Cache location: /.jenkins/cache/jars
Exception in thread "main" java.io.IOException: Failed to initialize FileSystem JAR Cache in /.jenkins/cache/jars
	at hudson.remoting.Engine.startEngine(Engine.java:268)
	at hudson.remoting.Engine.startEngine(Engine.java:227)
	at hudson.remoting.jnlp.Main.main(Main.java:228)
	at hudson.remoting.jnlp.Main._main(Main.java:223)
	at hudson.remoting.jnlp.Main.main(Main.java:189)
Caused by: java.lang.IllegalArgumentException: Root directory not writable: /.jenkins/cache/jars
	at hudson.remoting.FileSystemJarCache.<init>(FileSystemJarCache.java:62)
	at hudson.remoting.Engine.startEngine(Engine.java:266)
	... 4 more
Caused by: java.nio.file.AccessDeniedException: /.jenkins
	at sun.nio.fs.UnixException.translateToIOException(UnixException.java:84)
	at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:102)
	at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:107)
	at sun.nio.fs.UnixFileSystemProvider.createDirectory(UnixFileSystemProvider.java:384)
	at java.nio.file.Files.createDirectory(Files.java:674)
	at java.nio.file.Files.createAndCheckIsDirectory(Files.java:781)
	at java.nio.file.Files.createDirectories(Files.java:767)
	at hudson.remoting.Util.mkdirs(Util.java:159)
	at hudson.remoting.FileSystemJarCache.<init>(FileSystemJarCache.java:60)
	... 5 more

May 02, 2019 1:42:18 PM org.csanchez.jenkins.plugins.kubernetes.KubernetesLauncher launch
WARNING: Error in provisioning; agent=KubernetesSlave name: jenkins-slave-rzhdn, template=PodTemplate{inheritFrom='', name='jenkins-slave', slaveConnectTimeout=0, label='jenkins-slave', serviceAccount='jenkins', nodeSelector='', containers=[ContainerTemplate{name='jnlp', image='openshift/jenkins-slave-maven-centos7', workingDir='/tmp', command='', args='${computer.jnlpmac} ${computer.name}', resourceRequestCpu='', resourceRequestMemory='', resourceLimitCpu='', resourceLimitMemory=''}]}
java.lang.IllegalStateException: Containers are terminated with exit codes: {jnlp=1}
	at org.csanchez.jenkins.plugins.kubernetes.KubernetesLauncher.launch(KubernetesLauncher.java:156)
	at hudson.slaves.SlaveComputer$1.call(SlaveComputer.java:294)
	at jenkins.util.ContextResettingExecutorService$2.call(ContextResettingExecutorService.java:46)
	at jenkins.security.ImpersonatingExecutorService$2.call(ImpersonatingExecutorService.java:71)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
```

**Comment by gabemontero** [comment]
perhaps we need to do more `chmod` in the Dockerfiles @bparees  ... I'm investigating `/.jenkins` now ...

**Comment by gabemontero** [comment]
Ok, @bparees @adambkaplan this is how the jenkins remoting code determines that path:

`new File(System.getProperty("user.home"),".jenkins/cache/jars");`

Now, when I'm rsh'ed in a jenkins pod, I see this on the jenkins pid:

` -Duser.home=/var/lib/jenkins`

So I'm a little confused on how we get `/.jenkins/cache/jars` in that exception above 
vs. `/var/lib/jenkins/.jenkins/cache/jars`

I'm going to go through a slave scenario, leverage @adambkaplan 's k8s plugin retention policy, and look in both master and slave pods.

**Comment by gabemontero** [comment]
`/.jenkins/cache/jars` mystery solved in #forum-devex discussion 

crio is messing with $HOME evidently, and that is what is passed into `-Duser.home` ... see https://github.com/openshift/jenkins/blob/master/2/contrib/s2i/run#L479

File: 2/contrib/jenkins/jenkins-common.sh [hunk 1]
```diff
@@ -16,6 +16,9 @@ function generate_passwd_file() {
 
   if [ x"$USER_ID" != x"0" -a x"$USER_ID" != x"997" ]; then
 
+    # strip out the value that CRIO injected.
+    grep -v $USER_ID:x:$USER_ID /etc/passwd > /tmp/passwd
```
Comment (bparees) [inline 2/contrib/jenkins/jenkins-common.sh]: this grep is based on how crio injects the entry:
https://github.com/cri-o/cri-o/pull/2022/files#diff-cd575165f84eed10d10eb1460b0927b4R261

**Comment by gabemontero** [comment]
@bparees img build failed accessing masters ... bet the api.ci masters upgrade hit us

**Comment by gabemontero** [comment]
Still there:

```
WARNING: No Working Directory. Using the legacy JAR Cache location: /.jenkins/cache/jars
Exception in thread "main" java.io.IOException: Failed to initialize FileSystem JAR Cache in /.jenkins/cache/jars
	at hudson.remoting.Engine.startEngine(Engine.java:268)
	at hudson.remoting.Engine.startEngine(Engine.java:227)
	at hudson.remoting.jnlp.Main.main(Main.java:228)
	at hudson.remoting.jnlp.Main._main(Main.java:223)
	at hudson.remoting.jnlp.Main.main(Main.java:189)
Caused by: java.lang.IllegalArgumentException: Root directory not writable: /.jenkins/cache/jars
	at hudson.remoting.FileSystemJarCache.<init>(FileSystemJarCache.java:62)
	at hudson.remoting.Engine.startEngine(Engine.java:266)
	... 4 more
Caused by: java.nio.file.AccessDeniedException: /.jenkins
	at sun.nio.fs.UnixException.translateToIOException(UnixException.java:84)
	at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:102)
	at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:107)
	at sun.nio.fs.UnixFileSystemProvider.createDirectory(UnixFileSystemProvider.java:384)
	at java.nio.file.Files.createDirectory(Files.java:674)
	at java.nio.file.Files.createAndCheckIsDirectory(Files.java:781)
	at java.nio.file.Files.createDirectories(Files.java:767)
	at hudson.remoting.Util.mkdirs(Util.java:159)
	at hudson.remoting.FileSystemJarCache.<init>(FileSystemJarCache.java:60)
	... 5 more

May 02, 2019 4:02:46 PM org.csanchez.jenkins.plugins.kubernetes.KubernetesLauncher launch
WARNING: Error in provisioning; agent=KubernetesSlave name: jenkins-slave-l82xh, template=PodTemplate{inheritFrom='', name='jenkins-slave', slaveConnectTimeout=0, label='jenkins-slave', serviceAccount='jenkins', nodeSelector='', containers=[ContainerTemplate{name='jnlp', image='openshift/jenkins-slave-maven-centos7', workingDir='/tmp', command='', args='${computer.jnlpmac} ${computer.name}', resourceRequestCpu='', resourceRequestMemory='', resourceLimitCpu='', resourceLimitMemory=''}]}
java.lang.IllegalStateException: Agent is not connected after 100 attempts, status: Running
	at org.csanchez.jenkins.plugins.kubernetes.KubernetesLauncher.launch(KubernetesLauncher.java:190)
	at hudson.slaves.SlaveComputer$1.call(SlaveComputer.java:294)
	at jenkins.util.ContextResettingExecutorService$2.call(ContextResettingExecutorService.java:46)
	at jenkins.security.ImpersonatingExecutorService$2.call(ImpersonatingExecutorService.java:71)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
```

**Comment by bparees** [comment]
this can't work because CRIO is setting /etc/passwd back to 644 on us so we can't write to it at all.

**Pull request closed without merging**